src/core/client_channel/retry_throttle.cc

```diff
@@ -85,10 +98,10 @@ bool ServerRetryThrottleData::RecordFailure() {
   ServerRetryThrottleData* throttle_data = this;
   GetReplacementThrottleDataIfNeeded(&throttle_data);
   // We decrement milli_tokens by 1000 (1 token) for each failure.
-  const uintptr_t new_value =
-      static_cast<uintptr_t>(gpr_atm_no_barrier_clamped_add(
-          &throttle_data->milli_tokens_, gpr_atm{-1000}, gpr_atm{0},
-          static_cast<gpr_atm>(throttle_data->max_milli_tokens_)));
+  const intptr_t new_value = ClampedAdd<intptr_t>(
```

Comment:
Can you add some tests to test/core/client_channel/retry_throttle_test.cc to show that we properly handle the overflow cases?